### DISCUSSION TITLE
Train a time series data

### DISCUSSION THREAD

**Discussion by yanuarf** [azavea/raster-vision#2154] [Q&A]
I am working on image segmentation with time series data. I am using TemporalMultiRasterSource() for .tiff image and label files. 
![image](https://github.com/azavea/raster-vision/assets/17594124/65b7b9f7-54c4-4227-a22e-020681f1bba2)
![image](https://github.com/azavea/raster-vision/assets/17594124/7ce0d008-66d6-4821-ae1e-a5b2a8a83443)

When I combine them using scene() it looks successful, but when I sampling using SemanticSegmentationRandomWindowGeoDataset() it raise  error: OpenCV(4.8.0) /io/opencv/modules/imgproc/src/resize.cpp:3699: error: (-215:Assertion failed) !dsize.empty() in function 'resize'
![image](https://github.com/azavea/raster-vision/assets/17594124/f0adfa1b-3a89-4e2c-8406-b6d16fba3dd0)
![image](https://github.com/azavea/raster-vision/assets/17594124/e493a51f-c05e-4cb3-a922-0f6386ea9e13)
![image](https://github.com/azavea/raster-vision/assets/17594124/6b77749f-b197-429b-8ea5-8d1e63c77e0b)

I am sure there are no empty on nodata value in every label and image data. Is there any solution to do this?

**Comment by AdeelH** [comment]
How big are your TIFFs? Could it be that they are too small to sample a 1000x1000 chip from? Does it work with smaller values in `size_lims`?

**Reply by yanuarf** [reply]
The size of TIFFs file are 10256x14602 for each image and label. It cannot work also with smaller size_lims.

Oh and for the record, it was a success when I used single data.

**Reply by AdeelH** [reply]
I just noticed that you are passing a `RasterSource` as the label source to `Scene`. Two notes on that:
- You need to pass a `SemanticSegmentationLabelSource`. You can create one [like so](https://docs.rastervision.io/en/stable/usage/tutorials/reading_labels.html#Semantic-Segmentation---SemanticSegmentationLabelSource):
  ```py
  from rastervision.core.data import SemanticSegmentationLabelSource

  label_source = SemanticSegmentationLabelSource(
      label_raster_source, class_config=class_config)
  ```
- Raster VIsion does not currently support temporal labels. So the `RasterSource` you pass to `SemanticSegmentationLabelSource` must be non-temporal. In other words, the temporal use case currently supported is a single label (shape: HxW) for a time series input (shape: TxCxHxW).

**Reply by yanuarf** [reply]
I found that I can stack uri file without using `TemporalMultiRasterSource` like this :
`uri_image_sources = [
train_image_uri_1,
train_image_uri_2]`
`uri_label_sources = [
    train_label_uri_1,
    train_label_uri_2]`
Then using `SemanticSegmentationRandomWindowGeoDataset.from_uris` I call that into `image_uri` and `label_raster_uri`. It works, but is it the right method?

**Reply by AdeelH** [reply]
What is the shape of the data when you sample from the dataset? 

If you pass multiple URIs to a `RasterSource`, Raster Vision will combine them into a VRT using GDAL. It is intended for mosaicing adjacent images together; I am not sure what happens if the images you pass overlap completely.

**Reply by yanuarf** [reply]
So it is unlikely that the method will be accurate for multi-temporal data. The data sampled from the dataset is in tiff format for the image and the label.

How do I convert a single label vector (shape: HxW) into a time series input (shape: TxCxHxW)? I have labels in geojson or tiff form

**Reply by AdeelH** [reply]
>What is the shape of the data when you sample from the dataset?
>>The data sampled from the dataset is in tiff format for the image and the label.

I meant what do you get when you do 
```py
train_ds = SemanticSegmentationRandomWindowGeoDataset.from_uris(...)
x, y = train_ds[0]
x.shape, y.shape
```
while passing multiple image URIs to `image_uri`.

>How do I convert a single label vector (shape: HxW) into a time series input (shape: TxCxHxW)? I have labels in geojson or tiff form

Raster Vision expects labels to be HxW, so you don't need to change them at all.
